README.md

```diff
@@ -1,16 +1,715 @@
+| `ext` | `Any`                    | Non-normative extended fields                          | No       | `null`  |
+    
+### 3.1.1 Version
+
+The `v` field MUST contain the version of the invocation object  schema.
+
+### 3.1.2 Proofs
+
+The `prf` field MUST contain CIDs pointing to the UCANs that provide the authority to run these tasks. The elements of this array MUST be sorted in ascending numeric order. Restricting the outmost UCANs to only the authority required for the current invocation is RECOMMENDED.
+
+### 3.1.3 Run Capabilities
+
+The `run` field MUST reference the tasks contained in the UCAN are to be run during the invocation. To run all tasks in the underlying UCAN, the `"*"` value MUST be used. If only specific tasks (or [pipelines](#5-promise-pipelining)) are intended to be run, then they MUST be prefixed with an arbitrary label and treated as a UCAN attenuation: all tasks MUST be backed by a matching capability of equal or greater authority.
+
+#### 3.1.3.1 Promises
+
+The only difference from general capabilities is that [promises](#5-promise-pipelining) MAY also be used as inputs to attenuated fields.
+
+If a capability input has the key `"_"` and the value is a promise, the input MUST be discarded and only used for determining sequencing tasks.
+
+### 3.1.4 Nonce
+
+The `nnc` field MUST contain a unique nonce. This field exists to enable making the CID of each invocation unique. While this field MAY be an empty string, it is NOT RECOMMENDED.
+
+### 3.1.5 Extended Fields
+
+The OPTIONAL `ext` field MAY contain arbitrary data. If not present, the `ext` field MUST be interpreted as `null`, including for [signature](#315-signature).
+
+## 3.2 IPLD Schema
+
+``` ipldsch
+type SignedInvocation struct {
+  inv Invocation (rename "ucan/invoke") 
+  sig VarSig
+}
+
+type Invocation struct {
+  v   SemVer  -- Version
+  prf [&UCAN] -- Authority to run this invocation
+  run Scope   -- Which tasks to invoke
+  nnc String  -- Nonce
+  ext nullable Any (implicit null) -- Extended fields
+}
+
+type Scope union {
+  | All "*"
+  | {String : Task}
+}
+
+type Task struct {
+  with   URI 
+  do     Ability
+  inputs Any
+  meta   {String : Any}
+} 
+```
+
+## 3.3 JSON Examples
+
+### 3.3.1 Run All
+
+ ``` js
+{
+  "ucan/invoke": {
+    "v": "0.1.0",
+    "prf": [
+      {"/": "bafkreie2cyfsaqv5jjy2gadr7mmupmearkvcg7llybfdd7b6fvzzmhazuy"},
+      {"/": "bafkreibbz5pksvfjyima4x4mduqpmvql2l4gh5afaj4ktmw6rwompxynx4"}
+    ],
+    "run": "*"
+    "nnc": "abcdef",
+    "ext": null
+  },
+  "sig": {"/": { "bytes:": "bdNVZn_uTrQ8bgq5LocO2y3gqIyuEtvYWRUH9YT-SRK6v_SX8bjt-VZ9JIPVTdxkWb6nhVKBt6JGpgnjABpOCA" }}
+}
+```
+
+### 3.3.2 Promise Pipelines
+
+Promise pipelines are handled in more detail in [section 5](#5-promise-pipelining). In brief, they enable taking the output of a task that has not yet run as the input to another task. Here is a simple example:
```

Comment:
So, for resolving CID content (that may not be on the node executing the task), is that an inherent promise pipeline?

Reply:
I'd love to get your thoughts, but I currently think they should get no special treatment at the level of UCAN invocations! CIDs are understood as some kind of weird string when used as a value. Thoughts?

Reply:
Right. I think saying that they are may be good enough :).